test/gpb_parse_tests.erl

```diff
@@ -799,6 +816,22 @@ can_tolower_record_names_test() ->
                  output=msg2} %% .. and result msgs to be to-lower
       ]}] = do_process_sort_defs(Defs, [msg_name_to_lower]).
 
+can_to_snake_record_names_test() ->
+    {ok, Defs} = parse_lines(["message MsgName1 {required MsgName2   f1=1;}",
+                              "message MsgName2 {required uint32 g1=1;}",
+                              "service SvcName1 {",
+                              "  rpc req(MsgName1) returns (MsgName2) {};",
+                              "}",
+                              "extend MsgName1 { optional uint32 fm2=2; }"]),
+    [{{msg,msg_name_1}, [#?gpb_field{name=f1, type={msg,msg_name_2}},
+                   #?gpb_field{name=fm2}]},
+     {{msg,msg_name_2}, [#?gpb_field{name=g1}]},
+     {{service,svc_name_1},
+      [#?gpb_rpc{name=req,
+                 input=msg_name_1,  %% both argument ...
+                 output=msg_name_2} %% .. and result msgs to be to-lower
+      ]}] = do_process_sort_defs(Defs, [msg_name_to_snake_case]).
+
```

Comment:
Maybe it could be worth adding also unit tests for the oneof/map crash in #75, for example the below (this is what I used locally to verify the change)
```erlang
can_tolower_record_names_with_oneof_test() ->
    {ok, Defs} = parse_lines(["message Msg1 {",
                              "  oneof u {",
                              "    Msg1   a = 1;",
                              "    uint32 b = 2;",
                              "  }",
                              "}"]),
    [{{msg,msg1}, [#gpb_oneof{fields=[#?gpb_field{name=a,type={msg,msg1}},
                                      #?gpb_field{name=b}]}]}] =
        do_process_sort_defs(Defs, [msg_name_to_lower]).

can_tolower_record_names_with_map_test() ->
    {ok, Defs} = parse_lines(["message Msg1 {",
                              "  required uint32 f = 1;",
                              "}"
                              "message Msg2 {"
                              "  map<string,Msg1> m = 1;",
                              "}"]),
    [{{msg,msg1}, [#?gpb_field{}]},
     {{msg,msg2}, [#?gpb_field{type={map,string,{msg,msg1}}}]}] =
        do_process_sort_defs(Defs, [msg_name_to_lower]).
```